src/middleware/Logout.jsx

```diff
@@ -0,0 +1,17 @@
+import { useEffect } from 'react';
+import { Navigate } from 'react-router-dom';
+
+import { useAppDispatch } from 'hooks';
+import { logoutUser } from 'store/auth/thunk';
+
+export default function Logout() {
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    (async () => {
+      await dispatch(logoutUser());
+    })();
+  }, []);
```

Comment:
`await` has no effect here
```suggestion
  useEffect(() => {
    dispatch(logoutUser());
  }, []);
```